sound/soc/sof/ipc4-topology.c

```diff
@@ -1215,13 +1250,39 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		struct snd_sof_widget *pipe_widget;
 		struct sof_ipc4_pipeline *pipeline;
 
-		pipe_widget = swidget->spipe->pipe_widget;
-		pipeline = pipe_widget->private;
 		ipc4_copier = (struct sof_ipc4_copier *)swidget->private;
 		gtw_attr = ipc4_copier->gtw_attr;
 		copier_data = &ipc4_copier->data;
 		available_fmt = &ipc4_copier->available_fmt;
 
+		pipe_widget = swidget->spipe->pipe_widget;
+		pipeline = pipe_widget->private;
+
+		if (pipeline->use_chain_dma) {
+			u32 host_dma_id;
+			u32 fifo_size;
+
+			host_dma_id = platform_params->stream_tag - 1;
+			pipeline->msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_HOST_ID(host_dma_id);
+
+			/* Set SCS bit for S16_LE format only */
+			if (params_format(fe_params) == SNDRV_PCM_FORMAT_S16_LE)
+				pipeline->msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_SCS_MASK;
+
+			/*
+			 * Despite its name the bitfield 'fifo_size' is used to define DMA buffer
+			 * size. The expression calculates 2ms buffer size.
+			 */
+			fifo_size = 2 * ((params_rate(fe_params) * params_channels(fe_params) *
+					  params_physical_width(fe_params))/ 8000);
```

Comment:
missing space before / 

````
CHECK: spaces preferred around that '/' (ctx:VxW)
#383: FILE: sound/soc/sof/ipc4-topology.c:1277:
+					  params_physical_width(fe_params))/ 8000);
 					                                   ^
````

Reply:
Should there be a ROUND_UP in case we ever support 44.1 kHz?

Reply:
Probably, if we want to be sure the buffer size is longer than 2ms. It would also make sense to move the "2 *" inside the bracket to have the maximum precision in for the division.